var/spack/repos/builtin/packages/petsc/package.py

```diff
@@ -127,7 +134,9 @@ def install(self, spec, prefix):
                 'complex' if '+complex' in spec else 'real'),
             '--with-shared-libraries=%s' % ('1' if '+shared' in spec else '0'),
             '--with-debugging=%s' % ('1' if '+debug' in spec else '0'),
-            '--with-blas-lapack-dir=%s' % spec['lapack'].prefix
+            '--with-lapack-lib=%s' % spec['lapack'].lapack_static_lib,
```

Comment:
why this change? There is no static lib for MKL, so this bound to fail. I suggest to keep it as is for now.